lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2185,6 +2185,41 @@ constexpr ExecutionContextScope *g_no_exe_ctx = nullptr;
     return result;                                                             \
   } while (0)
 
+#define VALIDATE_AND_RETURN_EXPECTED(IMPL, REFERENCE, TYPE, EXE_CTX, ARGS,     \
+                                     FALLBACK_ARGS, DEFAULT)                   \
+  do {                                                                         \
+    FALLBACK(REFERENCE, FALLBACK_ARGS, DEFAULT);                               \
+    auto result = IMPL();                                                      \
+    if (!ModuleList::GetGlobalModuleListProperties()                           \
+             .GetSwiftValidateTypeSystem())                                    \
+      return result;                                                           \
+    if (!GetSwiftASTContext(nullptr))                                          \
+      return result;                                                           \
+    if (ShouldSkipValidation(TYPE))                                            \
+      return result;                                                           \
+    if ((TYPE) && !ReconstructType(TYPE))                                      \
+      return result;                                                           \
+    ExecutionContext _exe_ctx(EXE_CTX);                                        \
+    /* When in the error backstop the sc will point into the stdlib. */        \
+    if (auto *frame = _exe_ctx.GetFramePtr())                                  \
+      if (frame->GetSymbolContext(eSymbolContextFunction).GetFunctionName() == \
+          SwiftLanguageRuntime::GetErrorBackstopName())                        \
+        return result;                                                         \
+    auto swift_scratch_ctx_lock = SwiftScratchContextLock(                     \
+        _exe_ctx == ExecutionContext() ? nullptr : &_exe_ctx);                 \
+    bool equivalent =                                                          \
+        !ReconstructType(TYPE) /* missing .swiftmodule */ ||                   \
+        (Equivalent(llvm::expectedToStdOptional(std::move(result)),            \
+                    llvm::expectedToStdOptional(                               \
+                        GetSwiftASTContextFromExecutionContext(&_exe_ctx)      \
+                            ->REFERENCE ARGS)));                               \
+    if (!equivalent)                                                           \
+      llvm::dbgs() << "failing type was " << (const char *)TYPE << "\n";       \
+    assert(equivalent &&                                                       \
+           "TypeSystemSwiftTypeRef diverges from SwiftASTContext");            \
+    return IMPL();                                                             \
+  } while (0)
+
 #else
 #define VALIDATE_AND_RETURN_STATIC(IMPL, REFERENCE)                            \
   FALLBACK(REFERENCE, ());                                                     \
```

Comment:
This `FALLBACK(REFERENCE, ())` has not been modified for the third parameter.

Also the one in line 2229 which I cannot leave a comment in.

One also need `VALIDATE_AND_RETURN_EXPECTED` equivalent in that `#else` branch.

Reply:
https://github.com/apple/llvm-project/pull/8394